.github/workflows/master.yml

```diff
@@ -0,0 +1,260 @@
+        run: |
+          cd servicemodels/e2sm_ni
+          CGO_ENABLED=1 go build -o build/_output/e2sm_ni.so.1.0.0 -buildmode=plugin .
+  
+  build-e2sm_rc_pre:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v5
+        with:
+          go-version-file: 'go.mod'
+      - name: Build
+        run: |
+          cd servicemodels/e2sm_rc_pre
+          CGO_ENABLED=1 go build -o build/_output/e2sm_rc_pre.so.1.0.0 -buildmode=plugin .
+
+  build-e2sm_mho:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v5
+        with:
+          go-version-file: 'go.mod'
+      - name: Build
+        run: |
+          cd servicemodels/e2sm_mho
+          CGO_ENABLED=1 go build -o build/_output/e2sm_mho.so.1.0.0 -buildmode=plugin .
+
+  build-e2sm_rc:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v5
+        with:
+          go-version-file: 'go.mod'
+      - name: Build
+        run: |
+          cd servicemodels/e2sm_rc
+          CGO_ENABLED=1 go build -o build/_output/e2sm_rc.so.1.0.0 -buildmode=plugin .
+  
+  build-protoc_gen_cgo:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v5
+        with:
+          go-version-file: 'go.mod'
+      - name: Build
+        run: |
+          cd protoc-gen-cgo/
+          go build -v -o ./protoc-gen-cgo && cd ..
+
+  build-protoc_gen_choice:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v5
+        with:
+          go-version-file: 'go.mod'
+      - name: Build
+        run: |
+          cd protoc-gen-choice/
+          go build -v -o ./protoc-gen-choice && go install && cd ..
+
+  build-protoc_gen_builder:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v5
+        with:
+          go-version-file: 'go.mod'
+      - name: Build
+        run: |
+          cd protoc-gen-builder/
+          go build -v -o ./protoc-gen-builder && go install && cd ..
+  
+  unit-tests:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: golangci/golangci-lint-action@v3
+        with:
+          version: v1.54
+      - name: Unit tests
+        run: make test
+
+
+  sm-linters:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: golangci/golangci-lint-action@v3
+        with:
+          version: v1.54
+      - name: sm-linters
+        run: |
+          make sm-linters
+        
+  buflint:
+   runs-on: ubuntu-latest
+   steps:
+     - uses: actions/checkout@v4
+     - name: buflint 
+       env:
+         BUF_VERSION: 1.0.0   
+       run: |
+         sudo mkdir -p /go/src/github.com/onosproject
+         cd /go/src/github.com/onosproject
+         sudo git clone https://github.com/onosproject/onos-e2-sm.git
```

Comment:
Given that you are cloning the repo, I do not see where you are using the possible changes that may occur as part of the local/new changes. Any thoughts on this?

Reply:
Done